evalml/tests/pipeline_tests/test_component_graph.py

```diff
@@ -707,36 +709,53 @@ def test_component_graph_dataset_with_different_types():
              'DateTime': [DateTimeFeaturizer, 'OneHot.x'],
-             'Scaler': [StandardScaler, 'DateTime.x'],
+             'Text': [TextFeaturizer, 'DateTime.x'],
+             'Scaler': [StandardScaler, 'Text.x'],
              'Random Forest': [RandomForestClassifier, 'Scaler.x'],
              'Elastic Net': [ElasticNetClassifier, 'Scaler.x'],
              'Logistic Regression': [LogisticRegressionClassifier, 'Random Forest', 'Elastic Net']}
 
     X = pd.DataFrame({'column_1': ['a', 'b', 'c', 'd', 'a', 'a', 'b', 'c', 'b'],
                       'column_2': [1, 2, 3, 4, 5, 6, 5, 4, 3],
                       'column_3': [True, False, True, False, True, False, True, False, False]})
+    X['column_4'] = [str((datetime(2021, 5, 21, 12, 0, 0) + timedelta(minutes=5 * x))) for x in range(len(X))]
+    X['column_5'] = X['column_4']
     y = pd.Series([1, 0, 1, 0, 1, 1, 0, 0, 0])
-    X = infer_feature_types(X, {"column_2": "categorical"})
+    X = infer_feature_types(X, {"column_2": "categorical", "column_5": "NaturalLanguage"})
 
     component_graph = ComponentGraph(graph)
     component_graph.instantiate({})
     assert component_graph.input_feature_names == {}
     component_graph.fit(X, y)
 
     input_feature_names = component_graph.input_feature_names
-    assert input_feature_names['Imputer'] == ['column_1', 'column_2', 'column_3']
-    assert input_feature_names['OneHot'] == ['column_1', 'column_2', 'column_3']
-    assert input_feature_names['DateTime'] == ['column_3', 'column_1_a', 'column_1_b', 'column_1_c', 'column_1_d',
+    assert input_feature_names['Imputer'] == ['column_1', 'column_2', 'column_3', 'column_4', 'column_5']
+    assert input_feature_names['OneHot'] == ['column_1', 'column_2', 'column_3', 'column_4', 'column_5']
+    assert input_feature_names['DateTime'] == ['column_3', 'column_4', 'column_5',
+                                               'column_1_a', 'column_1_b', 'column_1_c', 'column_1_d',
                                                'column_2_1', 'column_2_2', 'column_2_3', 'column_2_4', 'column_2_5', 'column_2_6']
-    assert input_feature_names['Scaler'] == ['column_3', 'column_1_a', 'column_1_b', 'column_1_c', 'column_1_d',
-                                             'column_2_1', 'column_2_2', 'column_2_3', 'column_2_4', 'column_2_5', 'column_2_6']
-    assert input_feature_names['Random Forest'] == ['column_3', 'column_1_a', 'column_1_b', 'column_1_c', 'column_1_d',
-                                                    'column_2_1', 'column_2_2', 'column_2_3', 'column_2_4', 'column_2_5', 'column_2_6']
-    assert input_feature_names['Elastic Net'] == ['column_3', 'column_1_a', 'column_1_b', 'column_1_c', 'column_1_d',
-                                                  'column_2_1', 'column_2_2', 'column_2_3', 'column_2_4', 'column_2_5', 'column_2_6']
+    assert input_feature_names['Text'] == ['column_3', 'column_5', 'column_1_a', 'column_1_b', 'column_1_c', 'column_1_d',
```

Comment:
Woodwork would infer the data in "column_5" to be datetime, but I set it to natural language above. We check here that it passes through the datetime featurizer untouched and enters the text featurizer. On the next assert, we check that it gets transformed into natural language features.